modules/navier_stokes/src/physics/WCNSFVTurbulencePhysics.C

```diff
@@ -0,0 +1,328 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "WCNSFVTurbulencePhysics.h"
+#include "WCNSFVFlowPhysics.h"
+#include "WCNSFVFluidHeatTransferPhysics.h"
+#include "WCNSFVScalarTransportPhysics.h"
+#include "WCNSFVCoupledAdvectionPhysicsHelper.h"
+#include "NSFVAction.h"
+
+registerNavierStokesPhysicsBaseTasks("NavierStokesApp", WCNSFVTurbulencePhysics);
+registerMooseAction("NavierStokesApp", WCNSFVTurbulencePhysics, "add_variable");
+registerMooseAction("NavierStokesApp", WCNSFVTurbulencePhysics, "add_fv_kernel");
+registerMooseAction("NavierStokesApp", WCNSFVTurbulencePhysics, "add_aux_variable");
+registerMooseAction("NavierStokesApp", WCNSFVTurbulencePhysics, "add_aux_kernel");
+registerMooseAction("NavierStokesApp", WCNSFVTurbulencePhysics, "add_material");
+
+InputParameters
+WCNSFVTurbulencePhysics::validParams()
+{
+  InputParameters params = NavierStokesPhysicsBase::validParams();
+  params += WCNSFVCoupledAdvectionPhysicsHelper::validParams();
+  params.addClassDescription("Define a turbulence model for a weakly-compressible Navier Stokes "
+                             "flow with a finite volume discretization");
+
+  MooseEnum turbulence_type("mixing-length none", "none");
```

Comment:
Should also include k epsilon. Models are alreay merged and tested

Reply:
Will have to be in a follow up. 
It s bad practice to lump tons of work together, it makes reviewing harder